x-pack/plugins/observability_solution/infra/public/components/ml/anomaly_detection/anomalies_table/anomalies_table.tsx

```diff
@@ -119,13 +144,39 @@ const AnomalyActionMenu = ({
     closeFlyout();
   }, [jobId, onViewChange, startTime, type, influencers, influencerField, closeFlyout]);
 
+  const showInHostsList = useCallback(() => {
+    const newFilter = buildCombinedAssetFilter({
+      field: HOST_NAME_FIELD,
+      values: influencers,
+      dataView: metricsView?.dataViewReference,
+    });
+
+    if (!hostName || !closeFlyout) {
+      navigateToApp('metrics', {
+        replace: true,
+        path: `/hosts${location.state?.originSearch || ''}`,
+      }).then(() => {
+        addHostFilter(newFilter);
+      });
+    } else {
+      addHostFilter(newFilter);
+    }
+
+    if (closeFlyout !== undefined) closeFlyout();
```

Comment:
I did it initially but switched to undefined check because I wanted to be more specific. However, I agree that `if (closeFlyout)` is enough and more readable. Updated.